docs/upgrade.html

```diff
@@ -66,12 +66,23 @@
         individual messages is only reduced by the overhead of the batch format. This similarly affects the
         producer's <code>batch.size</code> configuration.</li>
     <li>GC log rotation is enabled by default, see KAFKA-3754 for details.</li>
+    <li>Message protocol introduces support for headers as an array of Header with uftf8 <code>key</code> and byte[] <code>value</code>. 
+        Header's will be available once the broker is upgraded to the latest protocol. Older clients will not be able to access the headers as they don't exist in previous protocols as such the broker removes the header on protocol downgrade.</li>
+    <li>ProducerRecord and ConsumerRecord client API's expose the new Headers API via <code>Headers headers()</code> method call. 
```

Comment:
It seems a little verbose to elaborate all the methods of Headers interface in the upgrade doc. See if the following is more concise.
```
<li>A new Headers interface has been introduced to provide headers read and write access.</li>
<li>ProducerRecord and ConsumerRecord expose the new Headers API via <code>Headers headers()</code> method call.</li>
<li>ExtendedSerializer and ExtendedDeserializer interfaces are introduced to support serialization and deserialization for headers. Headers will be ignored if the configured serializer and deserializer are not the above classes.</li>
```

There are some useful information in the current description, but may be more suitable for the Headers java doc. In the upgrade doc, it is probably not necessary to explain the details of a newly introduced feature (immutable headers after interceptor), but we do need to explain if there is any change of the existing behaviors.